### PR TITLE
932 - Fix filter issues with datagrid

### LINKED ISSUE #932: IdsDataGrid: Filter Improvements
**Describe the bug**
List of additional things to do with the filter row.

- [ ] https://github.com/infor-design/enterprise-wc/issues/667
- [x] Add full test coverage to `ids-data-grid.ts.html` or remove unneeded code
- [x] Improve Types
- [x] disableClientFilter setting (ignores filtering / ui only with a way to get the filter info by an event)
- [x] Confirm all the following work or have equivalents:  FilterTypes.Date, FilterTypes.Integer, FilterTypes.Decimal, FilterTypes.Dropdown, FilterTypes.Text, FilterTypes.Custom
- [x] On http://localhost:4300/ids-data-grid/filter.html the "In Stock" (boolean?) filter is not working

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the details for making this change. What existing problem does the pull request solve?**
Fixed some filter issues with datagrid

- Added support to disable client filter (ui only and get info by event)
- Added support to in-range for date filter
- More tests and coverage
- Update docs and other cleanup

**Related github/jira issue (required)**:
Fixes #932

**Steps necessary to review your pull request (required)**:
Pull this branch and build/run the demo app
- Navigate to:
- http://localhost:4300/ids-data-grid/filter.html
- http://localhost:4300/ids-data-grid/filter-custom.html
- http://localhost:4300/ids-data-grid/filter-trigger-fields.html
- http://localhost:4300/ids-data-grid/disable-client-filter.html (see console logs)
- Check filters and all tests should pass

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/ids-data-grid/README.md [hunk 1]
```diff
@@ -135,6 +135,9 @@ When used as an attribute in the DOM the settings are kebab case, when used in J
 - `columnGroups` {Array<object>} Allows you to group columns together in logical sets. See section below for details.
 - `rowHeight` {string | `'xs'` | `'sm'` | `'md'` | `'lg'`} Sets the height of each row
 - `data` {Array<object>} Sets the data to show in the data grid. This can be a JSON Array.
+- `disableClientFilter` {boolean} Sets to disable filter logic client side and let server do it.
+- `filterable` {boolean} Sets to turned on/off filters.
```
Comment (tmcconechy) [inline src/components/ids-data-grid/README.md:139]:
```suggestion
- `filterable` {boolean} Turns on or off the filter functionality.
```

File: src/components/ids-data-grid/README.md [hunk 2]
```diff
@@ -135,6 +135,9 @@ When used as an attribute in the DOM the settings are kebab case, when used in J
 - `columnGroups` {Array<object>} Allows you to group columns together in logical sets. See section below for details.
 - `rowHeight` {string | `'xs'` | `'sm'` | `'md'` | `'lg'`} Sets the height of each row
 - `data` {Array<object>} Sets the data to show in the data grid. This can be a JSON Array.
+- `disableClientFilter` {boolean} Sets to disable filter logic client side and let server do it.
```
Comment (tmcconechy) [inline src/components/ids-data-grid/README.md:138]:
```suggestion
- `disableClientFilter` {boolean} Disables the filter logic client side in situations you want to filter server side.
```

File: src/components/ids-data-grid/README.md [hunk 3]
```diff
@@ -135,6 +135,9 @@ When used as an attribute in the DOM the settings are kebab case, when used in J
...
+- `filterable` {boolean} Sets to turned on/off filters.
+- `filter-row-disabled` {boolean} Sets to disable filter row.
```
Comment (tmcconechy) [inline src/components/ids-data-grid/README.md:140]:
```suggestion
- `filter-row-disabled` {boolean} Disables the filter row.
```

File: src/components/ids-data-grid/README.md [hunk 4]
```diff
@@ -240,6 +243,7 @@ The formatter is then linked via the column on the formatter setting. When the g
 - `rowdeselected` Fires for each row that is deselected.
 - `rowactivated` Fires for each row that is activated.
 - `rowdeactivated` Fires for each row that is deactivated.
+- `clientfiltered` Fires after filter conditions changed, only fire with `disableClientFilter: true`.
```
Comment (tmcconechy) [inline src/components/ids-data-grid/README.md:246]:
```suggestion
- `clientfiltered` Fires after filter conditions are changed. Only fires when `disableClientFilter: true`.
```

File: src/components/ids-data-grid/README.md [hunk 5]
```diff
@@ -289,6 +293,7 @@ If the built-in filters are not enough, creating a custom filter is an option. T
 
 1. `UI Only` In order to do custom UI part of filter, add as html markup thru a slot. It must use slot and column-id attributes for example: `<div slot="filter-n" column-id="n">...</div>` where n is the columnId same passed in the columns.
 1. `filterFunction` This is a user defined filter method which must return a boolean. It determines if a cell value should be considered as a valid filtered value.
+1. `disableClientFilter` This is an api setting to disable filter logic client side. It will set filter conditions and fire an event `clientfiltered` which can listen for custom logic.
```
Comment (tmcconechy) [inline src/components/ids-data-grid/README.md:296]: Any reason to not use the same filtered event?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
A few small things...Also:

http://localhost:4300/ids-data-grid/filter-trigger-fields.html
- can center the active column
- for some reason the dropdown looks smaller?
Screen Shot 2022-10-21 at 10 10 48 AM

http://localhost:4300/ids-data-grid/filter-custom.html
- can center the in stock column
- can we show one with only a text input that works? This is an m3 use case

File: src/components/ids-data-grid/README.md (hunk 5 above)
Comment (deep7102) [reply src/components/ids-data-grid/README.md]: updated event name as `filtered`

**Comment by deep7102** [comment]
@tmcconechy updated now, all the points you find.

**Review by tmcconechy — APPROVED** [review]

**Review by andriivarhanov — APPROVED** [review]
looks good

**Pull request merged**